components/TaskList.vue

```diff
@@ -1,134 +1,41 @@
 <script setup lang="ts">
-import mockTasks from '@/mocks/tasks.js';
 
-const taskModal = ref(false);
-const taskInput = ref('');
-const removeTaskModal = ref(false);
-const taskIdToRemove = ref();
+const emit = defineEmits(['removeTask'])
 
-const props = defineProps({
-  'searchBy': {
-    type: String,
-    required: true,
-  }
-});
-
-const tasks = ref(mockTasks);
-
-const isTaskListPopulated = computed(() => tasks.value.length > 0);
-
-// watch(() => props.searchValue, (newValue) => {
-//   if (newValue) {
-//     return tasks.value.filter(task => task.title.toLowerCase().includes(props.searchValue.toLowerCase()));
-//   }
-// })
-
-const filteredTasks = computed(() => {
-  if (!props.searchBy) return tasks.value;
-
-  return tasks.value.filter(task =>
-    task.title.toLowerCase().includes(props.searchBy.toLowerCase())
-  );
-});
-
-const clearModal = () => {
-  taskModal.value = false;
-  taskInput.value = '';
-};
-
-const addTask = () => {
-  tasks.value.push({
-    title: taskInput.value,
-    isDone: false,
-    date: '24 May',
-    id: Math.random(),
-  });
+type Task = {
```

Comment:
It's better to keep types in a separate file. It gives the possibility to reuse them in other places